manifests-gen/providers.go

```diff
@@ -102,17 +104,52 @@ func (p *provider) writeProviderComponentsToManifest(fileName string, objs []uns
 	return os.WriteFile(path.Join(*manifestsPath, fileName), ensureNewLine(combined), 0600)
 }
 
-// writeProviderComponentsConfigmap allows to write provider components to the provider (transport) ConfigMap.
-func (p *provider) writeProviderComponentsConfigmap(fileName string, objs []unstructured.Unstructured) error {
-	combined, err := utilyaml.FromUnstructured(objs)
+// compressToZstd compressses bytes using zstd
+func (p *provider) compressToZstd(data []byte) (bytes.Buffer, error) {
+	var compressed bytes.Buffer
+
+	writer, err := zstd.NewWriter(&compressed, zstd.WithEncoderLevel(zstd.EncoderLevelFromZstd(19)))
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed to initialize zstd writer: %w", err)
+	}
+	defer writer.Close()
+
+	_, err = writer.Write(data)
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed to encode using zstd: %w", err)
+	}
+
+	err = writer.Close()
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("failed to close zstd writer: %w", err)
+	}
+
+	return compressed, nil
+}
+
+// writeProviderCustomizedComponents writes the customized infrastructure components to allow for code review
+func (p *provider) writeProviderCustomizedComponents(resourceMap map[resourceKey][]unstructured.Unstructured) error {
+	crds, err := utilyaml.FromUnstructured(resourceMap[crdKey])
 	if err != nil {
-		return fmt.Errorf("error converting unstructure object to YAML: %w", err)
+		return fmt.Errorf("error converting unstructured object to YAML: %w", err)
 	}
 
+	other, err := utilyaml.FromUnstructured(resourceMap[otherKey])
+	if err != nil {
+		return fmt.Errorf("error converting unstructured object to YAML: %w", err)
+	}
+
+	combined := utilyaml.JoinYaml(crds, other)
+
+	return os.WriteFile(path.Join(*basePath, "openshift", "infrastructure-components-openshift.yaml"), ensureNewLine(combined), 0600)
```

Comment:
Can we make this filename a constant and also add a comment similar to the one you left in the description re: it being used only for reviewing purposes? Thanks!